src/pydiode/send.py

```diff
@@ -54,10 +61,13 @@ async def read(self):
             # redirection.
             # https://stackoverflow.com/a/71627449
             if not self.stream_reader:
-                loop = asyncio.get_event_loop()
-                self.stream_reader = asyncio.StreamReader()
-                protocol = asyncio.StreamReaderProtocol(self.stream_reader)
-                await loop.connect_read_pipe(lambda: protocol, sys.stdin.buffer)
+                if platform == "win32":
+                    return sys.stdin.buffer.read(self.chunk_max_data_bytes)  # Sync read for Windows
```

Comment:
If you only tested using file redirection, then you actually haven't tested this branch of the logic. I don't think this part will work. Notice that the other branch uses await instead of return. Carefully read through the code in the other branch of the logic, and make sure you understand the difference between return and await.